pkg/watchdogd/watchdogd.go

```diff
@@ -71,153 +101,273 @@ func MonitorOops() error {
+		default:
+			r = OpResultInvalidOp
 		}
-		log.Println("watchdog: Armed")
+		c.Write([]byte{byte(r)})
+		c.Close()
+	}
+}
 
-	armed: // Loop while armed. SIGUSR1 to break.
-		for {
-			select {
-			case <-time.After(opts.KeepAlive):
-				doMonitors(opts.Monitors)
-				if err := wd.KeepAlive(); err != nil {
-					log.Printf("watchdog: Failed to keepalive: %v", err)
-					// Keep trying to pet until the watchdog times out.
-				}
-			case s := <-signals:
-				if s == unix.SIGUSR1 {
-					break armed
-				}
-			case <-ctx.Done():
-				return wd.Close()
-			}
+// setupListener sets up a new "unix" network listener for the daemon.
+func setupListener(uds string) (*net.UnixListener, func(), error) {
+	os.Remove(uds)
+
+	l, err := net.ListenUnix("unix", &net.UnixAddr{uds, "unix"})
+	if err != nil {
+		return nil, nil, err
+	}
+	cleanup := func() {
+		os.Remove(uds)
+	}
+	return l, cleanup, nil
+}
+
+// armWatchdog starts watchdog timer.
+func (d *Daemon) ArmWatchdog() rune {
+	if d.CurrentOpts == nil {
+		log.Printf("Current daemon opts is nil, don't know how to arm Watchdog")
+		return OpResultError
+	}
+	wd, err := watchdog.Open(d.CurrentOpts.Dev)
+	if err != nil {
+		// Most likely cause is /dev/watchdog does not exist.
+		// Second most likely cause is another process (perhaps
+		// another watchdogd?) has the file open.
+		log.Printf("Failed to arm: %v", err)
+		return OpResultError
+	}
+	if d.CurrentOpts.Timeout != nil {
+		if err := wd.SetTimeout(*d.CurrentOpts.Timeout); err != nil {
+			d.CurrentWd.Close()
+			log.Printf("Failed to set timeout: %v", err)
+			return OpResultError
 		}
-		if err := wd.MagicClose(); err != nil {
-			log.Printf("watchdog: Failed to disarm: %v", err)
-		} else {
-			log.Println("watchdog: Disarmed")
+	}
+	if d.CurrentOpts.PreTimeout != nil {
+		if err := wd.SetPreTimeout(*d.CurrentOpts.PreTimeout); err != nil {
+			d.CurrentWd.Close()
+			log.Printf("Failed to set pretimeout: %v", err)
+			return OpResultError
 		}
+	}
+	d.CurrentWd = wd
+	log.Printf("Watchdog armed")
+	return OpResultOk
+}
+
+// disarmWatchdog disarm the watchdog if already armed.
+func (d *Daemon) DisarmWatchdog() rune {
+	if d.CurrentWd == nil {
+		log.Printf("No armed Watchdog")
+		return OpResultOk
+	}
+	if err := d.CurrentWd.MagicClose(); err != nil {
+		log.Printf("Failed to disarm watchdog: %v", err)
+		return OpResultError
+	}
+	log.Println("Watchdog disarmed.")
+	return OpResultOk
+}
+
+// doPetting sends keepalive signal to Watchdog when necessary.
+//
+// If at least one of the custom monitors failed check(s), it won't send a keepalive
+// signal.
+func (d *Daemon) DoPetting() error {
+	if d.CurrentWd == nil {
+		return fmt.Errorf("no reference to any Watchdog")
+	}
+	if err := doMonitors(d.CurrentOpts.Monitors); err != nil {
+		return fmt.Errorf("won't keepalive since at least one of the custom monitors failed: %v", err)
+	}
+	if err := d.CurrentWd.KeepAlive(); err != nil {
+		return err
+	}
+	log.Println("Just pet Watchdog.")
```

Comment:
Remove this print. It is a bit verbose.

Reply:
Is there sth like `DEBUG` mode, that can be depend upon and only print verbose in that mode ? 

if os.get('DEBUG') {
   log.Println("Just pet Watchdog.")
}

I agree this can be annoying everywhere, but simply removing it is another extreme,, if thing go wrong, we'd need sth printed